code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectResourceManager.java

```diff
@@ -127,11 +107,23 @@ public Object listInternal(SchemaFactory schemaFactory, String type, Map<Object,
         return  projects;
     }
 
-    private void giveProjectAccess(Account project, Policy policy) {
+    private Account giveProjectAccess(Account project, Policy policy) {
+        if (project == null) {
+            throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+        }
+        if (!authDao.hasAccessToProject(project.getId(), policy.getAccountId(),
+                policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
+            throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+        }
+        if (!project.getState().equals(CommonStatesConstants.ACTIVE) && !authDao.isProjectOwner(project.getId(), policy.getAccountId(),
+                policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())){
+            throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+        }
         if (authDao.isProjectOwner(project.getId(), policy.getAccountId(), policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
             DataAccessor.fields(project).withKey(ObjectMetaDataManager.CAPABILITIES_FIELD).set(Arrays.asList("owner"));
         }
         policy.grantObjectAccess(project);
+        return project;
```

Comment:
Do we change project by any chance at line 125? If so, do you think it would make sense to do "return objectManager.reload(project)"?